onnxruntime/test/python/quantization/test_save_activations.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+# coding: utf-8
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""Testing save_activations module
+"""
+
+import tempfile
+import unittest
+from pathlib import Path
+
+import numpy as np
+import onnx
+from onnx import TensorProto, helper, numpy_helper
+
+import onnxruntime
+from onnxruntime.quantization.calibrate import CalibrationDataReader
+from onnxruntime.quantization.save_activations import aug_model_for_act_saving, run_collect_activations
+
+
+def generate_input_initializer(tensor_shape, tensor_dtype, input_name):
+    """
+    Helper function to generate initializers for test inputs
+    """
+    tensor = np.random.normal(0, 0.3, tensor_shape).astype(tensor_dtype)
+    init = numpy_helper.from_array(tensor, input_name)
+    return init
+
+
+class TestDataReader(CalibrationDataReader):
+    """Random Data Input Generator
+    """
+
+    def __init__(self):
+        self.preprocess_flag = True
+        self.enum_data_dicts = []
+        self.count = 2
+        self.input_data_list = []
+        for _ in range(self.count):
+            self.input_data_list.append(np.random.normal(0, 0.33, [1, 3, 1, 3]).astype(np.float32))
+
+    def get_next(self):
+        if self.preprocess_flag:
+            self.preprocess_flag = False
+            input_name = "input"
+            self.enum_data_dicts = iter([{input_name: input_data} for input_data in self.input_data_list])
+        return next(self.enum_data_dicts, None)
+
+    def rewind(self):
+        self.preprocess_flag = True
+
+
+class TestSaveActivations(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
```

Comment:
possible to use `setUp(self):` for this?  https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp

Reply:
I think future test cases in this class may share the same temp directory

Reply:
setUpClass will run once when the class initializes. The temp dir will not be cleaned for each test case, which may allow test cases to depend on each other. Was this intended?

Reply:
temp dir shared by different test cases, as long as no file name conflicts they should be ok. sharing temp dir creation and clean up also makes test run faster.

Reply:
I would encourage setting up for each test case to avoid the possibility of test cases having dependency on each other since each test case should be self-contained (and parallelizable). I wouldn't expect cleaning up after each test case to be very slow?

Reply:
How can they depend on each other when each test case would use its own files?

Reply:
They don't, but using a setUpClass allows for this probability and using setUp/tearDown to clean after each test case reduces the likelihood. imo setUp should be use unless there is a reason not to. What do you think?